### PR TITLE
planner: support reader operators cost detail

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/36962

Problem Summary:

### What is changed and how it works?

support reader operators cost detail including `BatchPointGet` / `IndexReader` / `IndexScan` / `TableReader` / `TableScan`

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: planner/core/plan_cost.go [hunk 1]
```diff
@@ -300,21 +307,27 @@ func (p *PhysicalTableReader) GetPlanCost(_ property.TaskType, option *PlanCostO
...
+		rowCount = getCardinality(p.tablePlan, costFlag)
+		p.planCost += rowCount * rowSize * netFactor
 		// net seek cost
-		p.planCost += estimateNetSeekCost(p.tablePlan)
+		netSeekCost = estimateNetSeekCost(p.tablePlan)
+		p.planCost += netSeekCost
 		// consider concurrency
-		p.planCost /= float64(p.ctx.GetSessionVars().DistSQLScanConcurrency())
+		p.planCost /= float64(sqlScanConcurrency)
 	case kv.TiFlash:
```
Comment (time-and-fate) [inline planner/core/plan_cost.go:331]: The `kv.TiFlash` case is not handled.

File: planner/core/plan_cost.go [hunk 2]
```diff
@@ -448,16 +467,17 @@ func (p *PhysicalTableScan) GetPlanCost(taskType property.TaskType, option *Plan
...
-		rowSize := math.Max(p.getScanRowSize(), 2.0) // to guarantee logRowSize >= 1
+		rowSize = math.Max(p.getScanRowSize(), 2.0) // to guarantee logRowSize >= 1
 		logRowSize := math.Log2(rowSize)
-		selfCost = getCardinality(p, costFlag) * logRowSize * scanFactor
+		rowCount = getCardinality(p, costFlag)
+		selfCost = rowCount * logRowSize * scanFactor
 
 		// give TiFlash a start-up cost to let the optimizer prefers to use TiKV to process small table scans.
 		if p.StoreType == kv.TiFlash {
 			selfCost += 2000 * logRowSize * scanFactor
```
Comment (time-and-fate) [inline planner/core/plan_cost.go:477]: Do we need to record this?

Comment (qw4990) [reply planner/core/plan_cost.go:477]: Better to record it as well. @Yisaer

File: planner/core/plan_cost.go (hunk 1 above)
Comment (Yisaer) [reply planner/core/plan_cost.go:331]: Optimize Tracer can't handle tiflash well for now, thus we don't cover tiflash currently.

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**